ui/app/src/Views/SilenceForm/Types.elm

```diff
@@ -4,209 +4,198 @@ module Views.SilenceForm.Types
-        }
-
-
 type alias Model =
-    { silence : Result String Silence
-    , form : SilenceForm
+    { form : SilenceForm
+    , silenceId : ApiData String
+    , alerts : ApiData (List Alert)
     }
 
 
 type alias SilenceForm =
     { id : String
-    , createdBy : Result ( String, String ) String
-    , comment : Result ( String, String ) String
-    , startsAt : Result ( String, String ) ( String, Time )
-    , endsAt : Result ( String, String ) ( String, Time )
-    , duration : Result ( String, String ) ( String, Time )
-    , matchers : List ValidatedMatcher
+    , createdBy : ValidatedField
+    , comment : ValidatedField
+    , startsAt : ValidatedField
+    , endsAt : ValidatedField
+    , duration : ValidatedField
+    , matchers : List MatcherForm
+    }
+
+
+type alias MatcherForm =
+    { name : ValidatedField
+    , value : ValidatedField
+    , isRegex : Bool
     }
 
 
 type SilenceFormMsg
     = UpdateField SilenceFormFieldMsg
-    | CreateSilence Silence
-    | PreviewSilence Silence
+    | CreateSilence
+    | PreviewSilence
     | AlertGroupsPreview (ApiData (List Alert))
     | FetchSilence String
     | NewSilenceFromMatchers (List Matcher)
     | NewSilenceFromMatchersAndTime (List Matcher) Time
     | SilenceFetch (ApiData Silence)
-    | SilenceCreate (ApiData String)
+    | SilenceCreate (ApiData SilenceId)
 
 
 type SilenceFormFieldMsg
     = AddMatcher
+    | DeleteMatcher Int
     | UpdateStartsAt String
+    | ValidateStartsAt
     | UpdateEndsAt String
+    | ValidateEndsAt
     | UpdateDuration String
+    | ValidateDuration
     | UpdateCreatedBy String
+    | ValidateCreatedBy
     | UpdateComment String
-    | DeleteMatcher Int
+    | ValidateComment
     | UpdateMatcherName Int String
+    | ValidateMatcherName Int
     | UpdateMatcherValue Int String
+    | ValidateMatcherValue Int
     | UpdateMatcherRegex Int Bool
 
 
-validateCreatedBy : String -> Result ( String, String ) String
-validateCreatedBy =
-    stringNotEmpty
+initSilenceForm : Model
+initSilenceForm =
+    { form = empty
+    , silenceId = Utils.Types.Initial
+    , alerts = Utils.Types.Initial
+    }
 
 
-stringNotEmpty : String -> Result ( String, String ) String
-stringNotEmpty string =
-    if String.isEmpty string then
-        Err ( string, "Should not be empty" )
-    else
-        Ok string
+toSilence : SilenceForm -> Maybe Silence
+toSilence { id, comment, matchers, createdBy, startsAt, endsAt } =
+    Result.map5
+        (\nonEmptyComment validMatchers nonEmptyCreatedBy parsedStartsAt parsedEndsAt ->
+            { nullSilence
+                | id = id
+                , comment = nonEmptyComment
+                , matchers = validMatchers
+                , createdBy = nonEmptyCreatedBy
+                , startsAt = parsedStartsAt
+                , endsAt = parsedEndsAt
+            }
+        )
+        (stringNotEmpty comment.value)
+        (List.foldr appendMatcher (Ok []) matchers)
```

Comment:
When I don't specify any matchers and create a silence, I get redirected to the `/silences` page without any error messages in the UI. The API call fails with `silence invalid: at least one matcher required`. Should we enforce having at least one matcher in this PR or in an upcoming one? Sorry, I missed this requirement in my last changes as well.